hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/ConsumerMessageSender.java

```diff
@@ -80,8 +83,9 @@ public void sendMessage(final Message message) {
         }
     }
 
-    public void updateSubscription(Subscription newSubscription) {
+    public synchronized void updateSubscription(Subscription newSubscription) {
         this.subscription = newSubscription;
+        this.messageSender = messageSenderFactory.create(newSubscription);
```

Comment:
I think this is a good idea to rebuilt sending component but we need some logic that determines whether this is actually necessary. What you did will work but I think is a little bit too simplistic. I think there is a value in not disrupting delivery flaw if possible.